driver/pparse.ml

```diff
@@ -79,9 +85,10 @@ let apply_rewriter magic fn_in ppx =
   end;
   fn_out
 
-let read_ast magic fn =
+let read_ast (type a) (kind : a ast_kind) fn : a =
   let ic = open_in_bin fn in
   try
+    let magic = magic_of_kind kind in
     let buffer = really_input_string ic (String.length magic) in
     assert(buffer = magic); (* already checked by apply_rewriter *)
     Location.input_name := input_value ic;
```

Comment:
To be consistent with the other changes, I think these `input_value` calls should be annotated as well